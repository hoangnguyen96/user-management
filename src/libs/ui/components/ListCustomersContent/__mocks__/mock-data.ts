import { USER_ROLE } from "../../../../constants";

export const LIST_CONTENT = [
  {
    fullName: "Jane Cooper",
    role: USER_ROLE.ADMIN,
    company: "Microsoft",
    phoneNumber: "(225) 555-0118",
    email: "jane@microsoft.com",
    country: "United States",
    status: true,
    username: "admin",
    password: "123456",
    id: "1",
  },
  {
    fullName: "Floyd Miles",
    role: USER_ROLE.CONSUMER,
    company: "Yahoo",
    phoneNumber: "(205) 555-0100",
    email: "floyd@yahoo.com",
    country: "Kiribati",
    status: false,
    username: "consumer1",
    password: "123456",
    id: "2",
  },
  {
    fullName: "Ronald Richards",
    role: USER_ROLE.CONSUMER,
    company: "Adobe",
    phoneNumber: "(302) 555-0107",
    email: "ronald@adobe.com",
    country: "Israel",
    status: false,
    username: "consumer2",
    password: "123456",
    id: "3",
  },
  {
    fullName: "Marvin McKinney",
    role: USER_ROLE.CONSUMER,
    company: "Tesla",
    phoneNumber: "(252) 555-0126",
    email: "marvin@tesla.com",
    country: "Iran",
    status: true,
    username: "consumer3",
    password: "123456",
    id: "4",
  },
  {
    fullName: "Jerome Bell",
    role: USER_ROLE.CONSUMER,
    company: "Google",
    phoneNumber: "(629) 555-0129",
    email: "jerome@google.com",
    country: "Réunion",
    status: true,
    username: "consumer4",
    password: "123456",
    id: "5",
  },
  {
    fullName: "Kathryn Murphy",
    role: USER_ROLE.CONSUMER,
    company: "Micro",
    phoneNumber: "(406) 555-0120",
    email: "kathryn@microsoft.com",
    country: "Réunion",
    status: true,
    username: "consumer5",
    password: "123456",
    id: "6",
  },
  {
    fullName: "Jacob Jones",
    role: USER_ROLE.CONSUMER,
    company: "Amazon",
    phoneNumber: "(208) 555-0112",
    email: "jacob@yahoo.com",
    country: "Brazil",
    status: true,
    username: "consumer6",
    password: "123456",
    id: "7",
  },
  {
    fullName: "Kristin Watson",
    role: USER_ROLE.CONSUMER,
    company: "Facebook",
    phoneNumber: "(704) 555-0127",
    email: "kristin@facebook.com",
    country: "Åland Islands",
    status: false,
    username: "consumer7",
    password: "123456",
    id: "8",
  },
  {
    fullName: "Kris Leo",
    role: USER_ROLE.CONSUMER,
    company: "Facebook",
    phoneNumber: "(704) 555-0112",
    email: "krisleo@facebook.com",
    country: "Sahara",
    status: false,
    username: "consumer7",
    password: "123456",
    id: "9",
  },
];
