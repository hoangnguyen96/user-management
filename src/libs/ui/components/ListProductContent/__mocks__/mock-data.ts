export const LIST_CONTENT_PRODUCT = [
  {
    name: "Eggs",
    price: 83,
    quantity: 70,
    status: false,
    image: "/src/libs/ui/images/img1.jpg",
    code: "POD1",
    id: "1",
    userId: "1",
  },
  {
    name: "Hamburger",
    price: 32,
    quantity: 69,
    status: false,
    image: "/src/libs/ui/images/img2.jpg",
    code: "POD2",
    id: "2",
    userId: "2",
  },
  {
    name: "Radio",
    price: 82,
    quantity: 66,
    status: true,
    image: "/src/libs/ui/images/img3.jpg",
    code: "POD3",
    id: "3",
    userId: "3",
  },
  {
    name: "Coffee",
    price: 49,
    quantity: 17,
    status: true,
    image: "/src/libs/ui/images/img4.jpg",
    code: "POD4",
    id: "4",
    userId: "4",
  },
  {
    name: "Hat",
    price: 42,
    quantity: 73,
    status: false,
    image: "/src/libs/ui/images/img5.jpg",
    code: "POD5",
    id: "5",
    userId: "5",
  },
  {
    name: "Volleyball",
    price: 47,
    quantity: 9,
    status: true,
    image: "/src/libs/ui/images/img6.jpg",
    code: "POD6",
    id: "6",
    userId: "6",
  },
  {
    name: "Honey",
    price: 39,
    quantity: 7,
    status: false,
    image: "/src/libs/ui/images/img7.jpg",
    code: "POD7",
    id: "7",
    userId: "6",
  },
  {
    name: "Tomate",
    price: 20,
    quantity: 44,
    status: true,
    image: "/src/libs/ui/images/img8.jpg",
    code: "POD8",
    id: "8",
    userId: "8",
  },
  {
    name: "KungFu",
    price: 21,
    quantity: 29,
    status: false,
    image: "/src/libs/ui/images/img9.jpg",
    code: "POD9",
    id: "9",
    userId: "8",
  },
  {
    name: "Karate",
    price: 59,
    quantity: 44,
    status: true,
    image: "/src/libs/ui/images/img10.jpg",
    code: "POD10",
    id: "10",
    userId: "8",
  },
  {
    name: "Bana",
    price: 6,
    quantity: 35,
    status: true,
    image: "/src/libs/ui/images/img10.jpg",
    code: "POD11",
    id: "11",
    userId: "6",
  },
];

export const LIST_CONTENT_PRODUCT_PAGE1 = [
  {
    name: "Eggs",
    price: 83,
    quantity: 70,
    status: false,
    image: "/src/libs/ui/images/img1.jpg",
    code: "POD1",
    id: "1",
    userId: "1",
  },
  {
    name: "Hamburger",
    price: 32,
    quantity: 69,
    status: false,
    image: "/src/libs/ui/images/img2.jpg",
    code: "POD2",
    id: "2",
    userId: "2",
  },
  {
    name: "Radio",
    price: 82,
    quantity: 66,
    status: true,
    image: "/src/libs/ui/images/img3.jpg",
    code: "POD3",
    id: "3",
    userId: "3",
  },
  {
    name: "Coffee",
    price: 49,
    quantity: 17,
    status: true,
    image: "/src/libs/ui/images/img4.jpg",
    code: "POD4",
    id: "4",
    userId: "4",
  },
  {
    name: "Hat",
    price: 42,
    quantity: 73,
    status: false,
    image: "/src/libs/ui/images/img5.jpg",
    code: "POD5",
    id: "5",
    userId: "5",
  },
  {
    name: "Volleyball",
    price: 47,
    quantity: 9,
    status: true,
    image: "/src/libs/ui/images/img6.jpg",
    code: "POD6",
    id: "6",
    userId: "6",
  },
  {
    name: "Honey",
    price: 39,
    quantity: 7,
    status: false,
    image: "/src/libs/ui/images/img7.jpg",
    code: "POD7",
    id: "7",
    userId: "6",
  },
  {
    name: "Tomate",
    price: 20,
    quantity: 44,
    status: true,
    image: "/src/libs/ui/images/img8.jpg",
    code: "POD8",
    id: "8",
    userId: "8",
  },
  {
    name: "KungFu",
    price: 21,
    quantity: 29,
    status: false,
    image: "/src/libs/ui/images/img9.jpg",
    code: "POD9",
    id: "9",
    userId: "8",
  },
  {
    name: "Karate",
    price: 59,
    quantity: 44,
    status: true,
    image: "/src/libs/ui/images/img10.jpg",
    code: "POD10",
    id: "10",
    userId: "8",
  },
];
export const LIST_CONTENT_PRODUCT_PAGE2 = [
  {
    name: "Bana",
    price: 6,
    quantity: 35,
    status: true,
    image: "/src/libs/ui/images/img10.jpg",
    code: "POD11",
    id: "11",
    userId: "6",
  },
];
