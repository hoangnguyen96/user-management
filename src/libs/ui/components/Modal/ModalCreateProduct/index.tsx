import { memo, useCallback } from "react";

// Models
import { Product, ProductInit, UserResponse } from "@app/models";

// Components
import ModalBase from "../ModalBase";
import FormProduct from "../../FormProduct";

interface ModalCreateProductProps {
  isOpen: boolean;
  isLoading: boolean;
  isAdmin?: boolean;
  idCustomer?: string;
  itemUpdate: Partial<Product>;
  listUser: UserResponse[];
  onClose: () => void;
  onSubmit: (payload: Partial<Product>) => void;
}

const ModalCreateProduct = ({
  isOpen,
  isLoading,
  isAdmin,
  idCustomer,
  itemUpdate,
  listUser,
  onClose,
  onSubmit,
}: ModalCreateProductProps) => {
  const { id } = itemUpdate;

  const handleSubmit = useCallback(
    (values: Partial<ProductInit>) => {
      const payload: Partial<Product> = {
        ...values,
        code: `POD0${values.userId || idCustomer}`,
        name: values.name,
        quantity: parseInt(values.quantity || ""),
        image: "/src/libs/ui/images/img9.jpg",
        price: isAdmin && values.price ? parseInt(values.price) : 0,
        userId: values.userId || idCustomer,
      };

      onSubmit(payload);
    },
    [id, onSubmit]
  );

  return (
    <ModalBase title="ADD PRODUCT" isOpen={isOpen} onClose={onClose}>
      {/* Form Update */}
      <FormProduct
        isAdmin={isAdmin}
        listUser={listUser}
        itemUpdate={itemUpdate}
        isLoading={isLoading}
        onSubmit={handleSubmit}
      />
    </ModalBase>
  );
};

export default memo(ModalCreateProduct);
