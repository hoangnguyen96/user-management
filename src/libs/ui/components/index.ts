// Common
export * from "./Common";

// Components
export { default as HeadingContent } from "./HeadingContent";
export { default as LazyFallbackContent } from "./LazyFallbackContent";
export { default as Logo } from "./Logo";
export { default as Footer } from "./Footer";
export { default as FormController } from "./FormController";
export { default as ListCustomersContent } from "./ListCustomersContent";
export { default as ListProductContent } from "./ListProductContent";
export { default as NavigationBar } from "./NavigationBar";
export { default as Temporary } from "./Temporary";
export { default as TitleContent } from "./TitleContent";
export { default as MenuBar } from "./MenuBar";
export { default as ErrorBoundary } from "./ErrorBoundary";
export { default as ModalCreateUserSuccess } from "./Modal/ModalCreateUserSuccess";
