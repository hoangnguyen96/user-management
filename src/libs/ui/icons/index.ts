export * from "./HelpIcon";
export * from "./IncomeIcon";
export * from "./KeyIcon";
export * from "./Members";
export * from "./ProductIcon";
export * from "./PromoteIcon";
export * from "./Screen";
export * from "./SearchIcon";
export * from "./TotalCustomers";
export * from "./UserIcon";
