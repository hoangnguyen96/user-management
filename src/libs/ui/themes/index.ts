import { createTheme } from "@mui/material/styles";
import { breakpointsBase } from "./breakpoints";
import { colors } from "./colors";
import { typographyBase } from "./typography";

export const themeDefault = createTheme({
  breakpoints: breakpointsBase,
  palette: colors,
  typography: typographyBase,

  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: "capitalize",
        },
      },
      variants: [
        {
          props: { variant: "contained" },
          style: {
            color: "White",
          },
        },
        {
          props: { variant: "activate" },
          style: {
            backgroundColor: "rgba(22, 192, 152, 0.38)",
            color: "#008767",
            border: "1px solid #00b087",
            "&:hover": {
              backgroundColor: "rgba(22, 192, 152, 0.62)",
            },
          },
        },
        {
          props: { variant: "inactivate" },
          style: {
            backgroundColor: "#ffc5c5",
            color: "#df0404",
            border: "1px solid #df0404",
            "&:hover": {
              backgroundColor: "rgb(253 170 170)",
            },
          },
        },
        {
          props: { variant: "normal" },
          style: {
            backgroundColor: " rgba(245, 245, 245, 1)",
            color: "rgba(64, 75, 82, 1)",
            border: "1px solid #eeeeee",
            "&:hover": {
              backgroundColor: "rgb(219 219 219)",
            },
          },
        },
      ],
      defaultProps: {
        size: "medium",
      },
    },

    MuiTextField: {
      styleOverrides: {
        root: ({ theme }) => ({
          "& .MuiOutlinedInput-root": {
            borderRadius: "12px",
            fontSize: "14px",
            backgroundColor: theme.palette.common.white,
            "& .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input": {
              padding: "12px",
            },
            "& fieldset": {
              borderColor: "#b5b7C0",
            },
            "&:hover fieldset": {
              borderColor: theme.palette.primary.main,
            },
            "&.Mui-focused fieldset": {
              borderColor: theme.palette.primary.main,
            },
          },
        }),
      },
    },

    MuiSelect: {
      styleOverrides: {
        root: {
          width: "auto",
          flex: 1,
        },
      },
    },

    MuiFormControl: {
      styleOverrides: {
        root: {
          flexDirection: "row",
          justifyContent: "center",
          alignItems: "center",
          backgroundColor: "#f9fbff",
          borderRadius: 12,
          minWidth: 154,
          "& .css-wama2x-MuiFormLabel-root-MuiInputLabel-root": {
            position: "unset",
            transform: "unset",
            left: "unset",
            top: "unset",
          },
          "& .css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root":
            {
              "& fieldset": {
                border: 0,
              },
              borderColor: "transparent",
              borderWidth: 0,
            },
          "& .css-wxw8j3-MuiFormLabel-root-MuiInputLabel-root": {
            position: "unset",
            transform: "unset",
            left: "unset",
            top: "unset",
            flex: 0.8,
            padding: 10,
            paddingRight: 5,
            fontSize: "12px",
            textAlign: "right",
          },
          "& .css-zbhh1g-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input.MuiSelect-select":
            {
              padding: "11px 32px 11px 0",
              fontSize: "12px",
              fontWeight: 500,
            },
          "& .css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root.Mui-focused .MuiOutlinedInput-notchedOutline":
            {
              borderColor: "transparent",
              borderWidth: 0,
            },
          "& .css-10d36jd-MuiSvgIcon-root-MuiSelect-icon, & .css-5ajf1d-MuiSvgIcon-root-MuiSelect-icon":
            {
              width: 20,
              height: 20,
            },
          "& .css-1j5jes1-MuiFormLabel-root.Mui-focused": {
            color: "rgba(0, 0, 0, 0.6)",
          },
        },
      },
    },

    MuiPaper: {
      styleOverrides: {
        root: {
          "& .css-ddqxy4-MuiButtonBase-root-MuiMenuItem-root": {
            fontSize: "12px",
          },
        },
      },
    },

    MuiListItemButton: {
      styleOverrides: {
        root: ({ theme }) => ({
          borderRadius: theme.spacing(1),
          "&.Mui-selected": {
            backgroundColor: theme.palette.primary.main,
            color: theme.palette.common.white,
          },
          "&.Mui-selected:hover": {
            backgroundColor: theme.palette.primary.main,
            color: theme.palette.common.white,
          },
        }),
      },
    },

    MuiLink: {
      styleOverrides: {
        root: ({ theme }) => ({
          backgroundColor: "transparent",
          fontSize: "14px",
          textDecoration: "none",
          color: theme.palette.primary.contrastText,
          "&:hover": {
            cursor: "pointer",
          },
        }),
      },
    },

    MuiPagination: {
      styleOverrides: {
        root: ({ theme }) => ({
          "& .MuiPagination-ul": {
            "& .MuiButtonBase-root.MuiPaginationItem-root": {
              fontSize: "12px",
              fontWeight: 500,
              backgroundColor: "#f5f5f5",
              minWidth: "25px",
              height: "24px",
              margin: "0 5px",
              "&.Mui-selected": {
                backgroundColor: theme.palette.primary.main,
                color: "white",
              },
              "& .MuiSvgIcon-root.MuiPaginationItem-icon": {
                fontSize: "12px",
              },
            },
          },
        }),
      },
    },

    MuiContainer: {
      styleOverrides: {
        root: {
          maxWidth: "100%",
          paddingLeft: 0,
          paddingRight: 0,
          "@media (min-width:600px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:960px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:1280px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:1440px)": {
            maxWidth: "1440px",
          },
        },
      },
    },
  },
});
