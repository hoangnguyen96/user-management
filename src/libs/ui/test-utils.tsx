import { PropsWithChildren, ReactElement, ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { BrowserRouter } from "react-router-dom";
import { RenderOptions, render } from "@testing-library/react";

// ThemeProvider
import { ThemeProvider } from "@mui/material";

// Config
import { themeDefault } from "./themes";

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <ThemeProvider theme={themeDefault()}>{children}</ThemeProvider>;
};

export const wrapper = ({ children }: { children: ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return (
    <ThemeProvider theme={themeDefault()}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderWithQueryClient = (
  ui: ReactElement,
  options?: RenderOptions
) => render(ui, { wrapper, ...options });

export const WrapperRouter = ({ children }: { children: ReactNode }) => (
  <BrowserRouter>{children}</BrowserRouter>
);

export { customRender as render };
export { customRenderWithQueryClient as renderWithQueryClient };
