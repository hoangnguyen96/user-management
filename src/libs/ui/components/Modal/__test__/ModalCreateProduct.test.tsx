import { renderWithQueryClient } from "@app/ui/test-utils";

// Component
import ModalCreateProduct from "../ModalCreateProduct";
import { LIST_CONTENT } from "../../ListCustomersContent/__mocks__/mock-data";

describe("ModalCreateProduct component", () => {
  const props = {
    isOpen: true,
    isLoading: false,
    itemUpdate: { name: "abc", quantity: 4 },
    listUser: LIST_CONTENT,
    onClose: () => {},
    onSubmit: () => {},
  };

  it("Should render snapshot correctly", () => {
    expect(
      renderWithQueryClient(<ModalCreateProduct {...props} />)
    ).toMatchSnapshot();
  });
});
