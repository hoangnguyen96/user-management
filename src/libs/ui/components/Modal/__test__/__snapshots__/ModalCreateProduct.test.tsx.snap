// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCreateProduct component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-1pvv9fj-MuiModal-root-MuiDialog-root"
      data-testid="close-modal"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-15jjlmt-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-l6nfb-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            ADD PRODUCT
          </h2>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-15cch7k-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <div
              class="MuiFormControl-root css-bcsigr-MuiFormControl-root"
            >
              <div
                class="MuiStack-root css-1l4xmop-MuiStack-root"
              >
                <div
                  class="MuiStack-root css-1lv7di1-MuiStack-root"
                >
                  <label
                    class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
                  >
                    Name:
                  </label>
                  <div
                    class="MuiStack-root css-aubdti-MuiStack-root"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1fihxl0-MuiFormControl-root-MuiTextField-root"
                      data-testid="name-product"
                    >
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          id=":r1:"
                          name="name"
                          placeholder="name"
                          type="text"
                          value="abc"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-ihdtdm"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiStack-root css-1lv7di1-MuiStack-root"
                >
                  <label
                    class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
                  >
                    Quantity:
                  </label>
                  <div
                    class="MuiStack-root css-aubdti-MuiStack-root"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root css-1fihxl0-MuiFormControl-root-MuiTextField-root"
                      data-testid="quantity-product"
                    >
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          id=":r2:"
                          inputmode="numeric"
                          name="quantity"
                          placeholder="1000"
                          type="text"
                          value="4"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-ihdtdm"
                          >
                            <span
                              class="notranslate"
                            >
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1bhl3b2-MuiButtonBase-root-MuiButton-root"
                data-testid="button-product"
                disabled=""
                tabindex="-1"
                type="button"
              >
                Add
                <div
                  class="MuiBox-root css-7u9dva"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
